internal/k8s/controllers/layer2_status_controller.go

```diff
@@ -99,40 +132,89 @@ func (r *Layer2StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	level.Debug(r.Logger).Log("controller", "Layer2StatusReconciler", "updated state", dumpResource(state))
-
 	return ctrl.Result{}, nil
 }
 
 func (r *Layer2StatusReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	p := predicate.NewPredicateFuncs(func(object client.Object) bool {
 		if s, ok := object.(*v1beta1.ServiceL2Status); ok {
-			return strings.HasSuffix(s.Name, r.NodeName)
+			// only objects with complete labels that can illustrate which service it is related to
+			// can trigger the reconciler
+			label := s.GetLabels()
```

Comment:
now that the status part is immutable we can remove anything related to labels in the predicate :)

Reply:
Please take a look at this, am I right about the predicates?https://github.com/metallb/metallb/pull/2351#discussion_r1616510392

Reply:
oh sorry I missed that, I guess you're right since `CreateOrPatch` updates the status subresource only when the object is there . I'm fine with leaving it on the labels then (at least for now)